src/platform_impl/linux/drm/event_loop.rs

```diff
@@ -0,0 +1,1417 @@
+                        file!(),
+                        crate::platform_impl::OsError::DrmMisc("Failed to open udev enumerator"),
+                    )
+                })?;
+                enumerator.match_subsystem("drm").map_err(|_| {
+                    crate::error::OsError::new(
+                        line!(),
+                        file!(),
+                        crate::platform_impl::OsError::DrmMisc("Failed to enumerate drm subsystem"),
+                    )
+                })?;
+                enumerator.match_sysname("card[0-9]*").map_err(|_| {
+                    crate::error::OsError::new(
+                        line!(),
+                        file!(),
+                        crate::platform_impl::OsError::DrmMisc("Failed to find a valid card"),
+                    )
+                })?;
+                enumerator
+                    .scan_devices()
+                    .map_err(|_| {
+                        crate::error::OsError::new(
+                            line!(),
+                            file!(),
+                            crate::platform_impl::OsError::DrmMisc("Failed to scan devices"),
+                        )
+                    })?
+                    .filter(|device| {
+                        let dev_seat_name = device
+                            .property_value("ID_SEAT")
+                            .map(|x| x.to_os_string())
+                            .unwrap_or_else(|| std::ffi::OsString::from("seat0"));
+                        if dev_seat_name == seat_name {
+                            if let Ok(Some(pci)) = device.parent_with_subsystem(Path::new("pci")) {
+                                if let Some(id) = pci.attribute_value("boot_vga") {
+                                    return id == "1";
+                                }
+                            }
+                        }
+                        false
+                    })
+                    .flat_map(|device| device.devnode().map(std::path::PathBuf::from))
+                    .next()
+                    .or_else(|| {
+                        enumerator
+                            .scan_devices()
+                            .ok()?
+                            .filter(|device| {
+                                device
+                                    .property_value("ID_SEAT")
+                                    .map(|x| x.to_os_string())
+                                    .unwrap_or_else(|| std::ffi::OsString::from("seat0"))
+                                    == seat_name
+                            })
+                            .flat_map(|device| device.devnode().map(std::path::PathBuf::from))
+                            .next()
+                    })
+                    .ok_or(crate::error::OsError::new(
+                        line!(),
+                        file!(),
+                        crate::platform_impl::OsError::DrmMisc("Failed to find suitable GPU"),
```

Comment:
Also, I think it'd make sense for that particular error when we failed to find GPU inform about env variable right away.

Reply:
fixed